### PR TITLE
Allow specifying the shard_transfer_mode when replicating reference tables

### PR DESCRIPTION
**Pull request by JelteF**
DESCRIPTION: Support logical replication in replicate_reference_tables()

When using `citus.replicate_reference_tables_on_activate = off`,
reference tables need to be replicated later. This can be done using the
`replicate_reference_tables()` UDF. However, this function only allowed
blocking replication. This changes the function to default to logical
replication instead, and allows choosing any of our existing shard
transfer modes.

### CODE REVIEW

**Review by marcocitus — APPROVED** [review]
yay! should it have a downgrade script?

File: src/backend/distributed/utils/reference_table_utils.c [hunk 1]
```diff
@@ -62,7 +62,9 @@ PG_FUNCTION_INFO_V1(replicate_reference_tables);
 Datum
 replicate_reference_tables(PG_FUNCTION_ARGS)
 {
-	EnsureReferenceTablesExistOnAllNodes();
+	Oid shardReplicationModeOid = PG_GETARG_OID(0);
+	char shardReplicationMode = LookupShardTransferMode(shardReplicationModeOid);
+	EnsureReferenceTablesExistOnAllNodesExtended(shardReplicationMode);
```
Comment (onderkalaci) [inline src/backend/distributed/utils/reference_table_utils.c:67]: This seems like a very important change.

 Do we have any existing tests that shows non-blocking `EnsureReferenceTablesExistOnAllNodesExtended`? Would be good to reference in the PR description.

Comment (marcocitus) [reply src/backend/distributed/utils/reference_table_utils.c:67]: rebalance_table_shards already calls the non-blocking variant by default, but it probably does make sense to cover the UDF call in tests

Comment (onderkalaci) [reply src/backend/distributed/utils/reference_table_utils.c:67]: > rebalance_table_shards already calls the non-blocking variant by default, 

Ah, cool, I missed that

Comment (JelteF) [reply src/backend/distributed/utils/reference_table_utils.c:67]: We had some tests for `replicate_reference_tables()` already, I modified those slightly to test all the shard transfer modes.

**Pull request merged**